### PR TITLE
Refactor out common method for matching regexps against a pool name.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Refactor some redundant code for matching pools against a list of regexps.

## Why are we making these changes?
Remove redundant code that will be used a third time.

### CODE REVIEW

File: scheduler/src/cook/tools.clj [hunk 1]
```diff
@@ -894,6 +894,15 @@
     (cond-> {:count 1 :cpus cpus :mem mem}
       gpus (assoc :gpus gpus))))
 
+(defn match-based-on-pool-name
```
Comment (nsinkov) [inline scheduler/src/cook/tools.clj:897]: add a unit test?

Comment (scrosby) [reply scheduler/src/cook/tools.clj:897]: There is one via (get-default-container-for-pool)

Comment (nsinkov) [reply scheduler/src/cook/tools.clj:897]: nitpick: I would make it separate still

Comment (scrosby) [reply scheduler/src/cook/tools.clj:897]: Ok. I added one. :)

**Pull request merged**